// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.proto";
option java_outer_classname = "DelayAsyncScriptExecutionMetadataProto";

package optimization_guide.proto;

// Enumerates the timings at which execution of async scripts can be delayed
// until.
enum DelayType {
  DELAY_TYPE_UNKNOWN = 0;
  DELAY_TYPE_FINISHED_PARSING = 1;
  DELAY_TYPE_FIRST_PAINT_OR_FINISHED_PARSING = 2;
};

// Optimization metadata for OptimizationType::DELAY_ASYNC_SCRIPT_EXECUTION.
message DelayAsyncScriptExecutionMetadata {
  // The timing at which execution of async scripts should be delayed until.
  optional DelayType delay_type = 1;
}
